### PR TITLE
bug: Allow text/plain, application/json, or no content type to be accepted

### PR DESCRIPTION
**Pull request by jrconlin**
Closes 115

### CODE REVIEW

**Comment by pjenvey** [comment]
> Closes 115

it's #113

File: src/web/extractors.rs [hunk 1]
```diff
@@ -285,7 +286,11 @@ impl FromRequest<ServerState> for BsoBody {
         }
         let mut config = JsonConfig::default();
         let max_request_size = req.state().limits.max_request_bytes as usize;
-        config.limit(max_request_size);
+        config.limit(max_request_size).content_type(|ct| {
+            ct.to_string() == "".to_owned()
+                || ct == mime::TEXT_PLAIN
```
Comment (pjenvey) [inline src/web/extractors.rs]: nit: I think we only need this line (at least according to test_storage.py). The mime crate is strongly typed and doesn't appear to parse an empty string content_type. Also JsonConfig always allows application/json

**Comment by pjenvey** [comment]
Unfortunately @philbooth's content_type feature somehow was lost in [actix-web 1.0](https://docs.rs/actix-web/1.0.0/actix_web/web/struct.JsonConfig.html), so we'll also need to redo that PR :(

File: src/web/extractors.rs (hunk 1 above)
Comment (jrconlin) [reply src/web/extractors.rs]: Honestly, I was being super paranoid by that point. My test worked when I didn't include `mime::APPLICATION_JSON`, but I had no idea if that was a fluke or not.

**Comment by jrconlin** [comment]
Also filed https://github.com/actix/actix-web/issues/894

**Pull request merged**